### PR TITLE
ALEC-75: Upgrade Alec Kafka Components to 2.2.0 on Scala 2.12

### PR DESCRIPTION
**Pull request by Naicisum**
ALEC-75: Upgrade Alec Kafka Components to 2.2.0 on Scala 1.12
ALEC-75: Cleanup opennms-kafka pom
ALEC-75: Update/Centralize dependency versions to reduce conflicts

### External References

* JIRA (Issue Tracker): https://issues.opennms.org/browse/ALEC-75

### CODE REVIEW

File: karaf-features/src/main/resources/features.xml [hunk 1]
```diff
@@ -238,14 +238,14 @@
...
     </feature>
 
-    <feature name="kafka-streams" description="Kafka Streams" version="2.0.0">
+    <feature name="kafka-streams" description="Kafka Streams" version="${kafka.version}">
         <feature version="5.7.3" dependency="true">rocksdb</feature>
-        <bundle dependency="true">mvn:org.apache.servicemix.bundles/org.apache.servicemix.bundles.kafka-streams/2.0.0_1</bundle>
-        <bundle dependency="true">mvn:org.apache.servicemix.bundles/org.apache.servicemix.bundles.kafka-clients/2.0.0_1</bundle>
+        <bundle dependency="true">mvn:org.apache.servicemix.bundles/org.apache.servicemix.bundles.kafka-streams/${kafka.bundle.version}</bundle>
+        <bundle dependency="true">mvn:org.apache.servicemix.bundles/org.apache.servicemix.bundles.kafka-clients/${kafka.bundle.version}</bundle>
         <bundle dependency="true">mvn:org.lz4/lz4-java/1.4.1</bundle>
```
Comment (j-white) [inline karaf-features/src/main/resources/features.xml]: We should also update the versions on the transient dependencies to match those use by 2.2.0. For example, 2.2.0 now depends on `lz4-java` 1.5.0.

Comment (Naicisum) [reply karaf-features/src/main/resources/features.xml]: Done

**Comment by Naicisum** [comment]
I worked on updating dependencies that were causing conflicts on the compile target using:
`mvn dependency:tree -Dverbose | grep conflict`

Conflicts on `compile` target have been resolved

Unfortunately the test suite `org.springframework.kafka:spring-kafka-test` is built on Scala 2.11 so it will always conflict on target `test`

Updating dependencies `selenium` in `smoke-test` removed another large portion of conflicts.

**Comment by j-white** [comment]
Merged into `release-1.x`.

**Pull request closed without merging**